### DISCUSSION TITLE
Local update server

### DISCUSSION THREAD

**Discussion by kadaan** [doudar/SmartSpin2k#90] [Q&A] [answered]
@doudar Can you think of a good way to have local update server for testing?  I was hoping for a way to be able to upload new firmware without having to change: https://github.com/doudar/SmartSpin2k/blob/6a01e1e853d0a06daa309f051c5e6cadbb90e1b1/include/settings.h#L24

**Comment by doudar** [comment, marked as answer]
Without changing, you'd have to setup a DNS server to resolve that to the local IP of a server hosting that file. 

For your own testing, it would probably be easiest if you just put the spiffs.bin and firmware.bin and version.txt from the .pio/build folder into your own OTAUpdates repository on github and point to the files with FW_UPDATEURL

**Comment by doudar** [comment]
Also, you bring up a great point I'll add to Issues. The [root CA certificate](https://github.com/doudar/SmartSpin2k/blob/master/include/cert.h) is associated with Github. That would need to be changed if the files aren't hosted by github or the https request would fail.

**Comment by kadaan** [comment]
@doudar Did I miss instructions on how to upload the firmware.bin and spiffs.bin to SmartSpin2k after building locally?

**Reply by doudar** [reply]
I sent them to you via Email

**Reply by kadaan** [reply]
I see it now.  Thanks!

**Reply by runty** [reply]
Can I get this as well please!?

**Reply by kadaan** [reply]
The best it to upload from within platformio in vscode. See: https://github.com/doudar/SmartSpin2k#if-you-are-a-developer

Rather than host an update server, it was easier to just upload the builds through platformio
